src/lib/utils/getPlainText.ts

```diff
@@ -0,0 +1,43 @@
+import type { SerializedEditorState } from '@payloadcms/richtext-lexical/lexical'
+
+type Node = {
+  type: string
+  text?: string
+  children?: Node[]
+}
+
+/**
+ *Gives you a plain-text string from a Lexical-serialized state (i.e RichText).
+ * @param data - The Lexical-serialized state to convert.
+ * @param maxLength - Optional maximum length. If provided, truncates and adds an ellipsis.
+ */
+export function getPlainText(
+  data: SerializedEditorState,
+  maxLength?: number,
+): string {
+  // recursive function to walk the node tree.
+  function walk(nodes: Node[]): string {
+    let out = ''
+    for (const n of nodes) {
+      if (n.type === 'text' && n.text) {
+        out += n.text
+      } else if (n.children) {
+        out += walk(n.children)
+      }
+      if (n.type === 'paragraph') {
+        out += '\n'
+      }
+    }
+    return out
+  }
+
+  // this will recursively retrieve all text from the RichText element.
+  const children = (data as any).root.children as Node[]
+  const full = walk(children).trim()
+
+  // this if statement will run if a maximum size for the plain text is passed as an arg.
+  if (typeof maxLength === 'number' && full.length > maxLength) {
+    return full.slice(0, maxLength).replace(/\s+$/, '') + '…'
+  }
+  return full
```

Comment:
```suggestion
  // this if statement will run if a maximum size for the plain text is passed as an arg.
  if (maxLength === undefined) return full
  
  if (full.length > maxLength) {
    return full.slice(0, maxLength).replace(/\s+$/, '') + '…'
  }
```

maybe doing an early return here is a bit cleaner? and then typescript can narrow down the type to be a number. (we could also do `if (!maxLength)` but this doesn't work for the edge case where `maxLength = 0`, depending on what you define the behavior as. 

On that thought, maybe we want to define some behaviour & tests for if maxLength= 0 and < 0?

Reply:
Tweaked if else flow a bit